app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :email, presence: true
+  validates :uid, presence: true
+
+  def self.from_omniauth(auth_info)
+    user = User.find_or_create_by(uid: auth_info[:uid])
```

Comment:
I believe this is not working as intended. I logged in a few times with my account and had 3 different user accounts

Reply:
We added uniqueness validations to the model and that seems to have fixed it. Good catch!